ui/app/src/services/content.ts

```diff
@@ -398,8 +401,43 @@ function writeContentUrl(qs: object) {
   return `/studio/api/1/services/api/1/content/write-content.json?${qs.toString()}`;
 }
 
-export function updateField(modelId: string, fieldId: string, value: any): Observable<any> {
-  throw new Error('Not implemented.');
+export function updateField(
+  site: string,
+  modelId: string,
+  fieldId: string,
+  indexToUpdate: number,
+  parentModelId: string = null,
+  value: any
+): Observable<any> {
+  return performMutation(
+    site,
+    modelId,
+    parentModelId,
+    doc => {
+      let fieldNode;
+      if (typeof indexToUpdate === 'string') {
```

Comment:
This logic is not sufficient for all cases of update.

For example, see header content type. Scenario: updating social_media.icon[0]. You should probably use extractNode in either case. And it may need a little tweaking to support.